src/power_shell.erl

```diff
@@ -153,7 +153,7 @@ eval_apply(false, Mod, Fun, Args, FunMap0) ->
             % recover stack from process dictionary
             erlang:raise(Class, Reason, get_stack() ++ tl(Trace))
     after
-        erase(?STACK_TOKEN)
+        put(?STACK_TOKEN, PreservedStack)
```

Comment:
This will leave unexpected `undefined` added to process dictionary when no previous stack was preserved.

To fix this, a test case needs to be amended to check via `process_info(self(), dictionary)` that ?STACK_TOKEN is not set (as opposed to set to 'undefined').

And here, it should be similar to:
```
    case PreservedStack of
        undefined -> erase(?STACK_TOKEN);
        List when is_list(List) -> put(?STACK_TOKEN, List)
```

Reply:
`erlang:get/1` returns `undefined` if no value has been set:
https://www.erlang.org/doc/man/erlang.html#get-1

So setting undefined, and erasing the key from the process dictionary will result in the same `get` return value. With the difference, that `get/0` will return `[{?STACK_TOKEN, undefined}]`.

Reply:
> With the difference, that get/0 will return [{?STACK_TOKEN, undefined}].

Yes - and that'd happen in the calling process context, which `power_shell` should keep unmodified.